core/server/worker/src/test/java/alluxio/worker/grpc/ShortCircuitBlockWriteHandlerTest.java

```diff
@@ -0,0 +1,300 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.worker.grpc;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
+import alluxio.grpc.CreateLocalBlockRequest;
+import alluxio.grpc.CreateLocalBlockResponse;
+import alluxio.util.io.PathUtils;
+import alluxio.worker.block.CreateBlockOptions;
+import alluxio.worker.block.NoopBlockWorker;
+
+import com.amazonaws.annotation.NotThreadSafe;
+import com.google.common.collect.ImmutableList;
+import io.grpc.stub.StreamObserver;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class ShortCircuitBlockWriteHandlerTest {
+
+  @Rule
+  public TemporaryFolder mTempFolder = new TemporaryFolder();
+
+  private TestResponseObserver mResponseObserver;
+  private TestBlockWorker mBlockWorker;
+  private ShortCircuitBlockWriteHandler mHandler;
+
+  @Before
+  public void before() throws Exception {
+    File tmpDir = mTempFolder.newFolder();
+    mBlockWorker = new TestBlockWorker(tmpDir.getAbsolutePath());
+
+    mResponseObserver = new TestResponseObserver();
+
+    mHandler = new ShortCircuitBlockWriteHandler(mBlockWorker, mResponseObserver);
+  }
+
+  @Test
+  public void createBlock() {
+    long blockId = 1L;
+    CreateLocalBlockRequest request = CreateLocalBlockRequest
+            .newBuilder()
+            .setBlockId(blockId)
+            .build();
```

Comment:
looks like you're creating this request again and again,  how about construct it in the before() method?